### ISSUE TITLE
Upload folders without having to sync an entire (sub)library first

### ISSUE THREAD

**Issue by xarinatan** [haiwen/seafile#1594]
Ever since I got seafile I've been absolutely addicted to it, it's my main fileserver for daily home and professional use. However, there's one thing I'm hugely missing in this daily usage, and that is the ability to upload folders without having to sync an entire library first. 
Dropping a folder on either the webinterface or the client's cloud browser doesn't work. The webinterface will attempt to upload all files in all subfolders, and put them in the current open folder (which isn't very desirable nor even possible with 30GB+ folders with hundreds of thousands of files). And the cloud browser in the client will just say 'Feature not supported'. 
Currently the only workaround I have, is to create a subfolder and then create a sublibrary out of that subfolder, then sync that folder and drop the files in there, but that's both inconvient and slow as well as clutters up the list of libraries.

In short, would it be possible to add the ability to upload entire folders with their folder structure in tact, without having to sync a (sub)library to my computer first? I **really** miss that feature.

**Comment by shoeper** [comment]
In the pro edition folders can be uploaded using Seahub (afaik this does work in chrome but not firefox - clicking on upload allows you to select "upload" folder in supported browsers). It keeps the structure but has limitations relating the file count (somewhere around 500 files I think).
Here it don't see much optimization potential because allowing much more files could lead to crashing the browser.

Another option (also pro edition) create a folder using the cloud file explorer and selectively sync it with your local folder (make a right click on the folder -> sync this folder). Once it finishes you can unsync it. This way works for folder of any size.

You already described these ways but I'm not sure if you were aware of the option to directly sync the subfolder without creating a sublibrary manually first.

One option I'd see for a small optimization would be an option to upload a folder to a library using the client. But it would just save the steps to create the folder, sync it and unsync it and instead would have a dialog like syncing a library (but for a folder).

If you think the create folder and sync subfolder option is not what you're searching for please add your feature request to the wishlist @ forum.seafile-server.org.
